src/values/AbstractValue.js

```diff
@@ -348,18 +305,39 @@ export default class AbstractValue extends Value {
     }
 
     let realm = this.$Realm;
+    if (realm.pathConditions.length === 0) return this;
```

Comment:
Why? Is this a performance optimization?

Reply:
Yes, but mostly it makes debugging less painful.